pkg/networkservice/common/clientinfo/client.go

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package clientinfo provides a chain element that adds pod, node and cluster names to request
+package clientinfo
+
+import (
+	"context"
+	"os"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
+)
+
+const (
+	nodeNameEnv      = "NODE_NAME"
+	podNameEnv       = "POD_NAME"
+	clusterNameEnv   = "CLUSTER_NAME"
+	nodeNameLabel    = "NodeNameKey"
+	podNameLabel     = "PodNameKey"
+	clusterNameLabel = "ClusterNameKey"
+)
+
+type clientInfo struct{}
+
+// NewClient - creates a new networkservice.NetworkServiceClient chain element that adds pod, node and cluster names
+// to request from corresponding environment variables
+func NewClient() networkservice.NetworkServiceClient {
+	return &clientInfo{}
+}
+
+func (a *clientInfo) Request(ctx context.Context, request *networkservice.NetworkServiceRequest, opts ...grpc.CallOption) (*networkservice.Connection, error) {
+	var names = map[string]string{
+		nodeNameEnv:    nodeNameLabel,
+		podNameEnv:     podNameLabel,
+		clusterNameEnv: clusterNameLabel,
+	}
+
+	conn := request.GetRequestConnection()
+	if conn.Labels == nil {
+		conn.Labels = make(map[string]string)
+	}
+	for envName, labelName := range names {
```

Comment:
Can you extract whole this labels manipulation as a function to a separate module, so it could be used also in implementation of https://github.com/networkservicemesh/sdk/issues/44 ?
I see that as a func which is getting a map of labels as an arg and returns nothing.
This will allow also to have common test for the same functionality used in two (there and in registry client) places.

Reply:
Ok, did it